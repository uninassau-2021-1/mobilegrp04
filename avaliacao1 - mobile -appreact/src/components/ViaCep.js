import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";

function ViaCepHook(props) {
  const [data, setData] = useState(null);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  useEffect(() => {
    if (props.lazy) return;
    getCep();
  }, [props.lazy]);

  function getCep() {
    setLoading(true);
    fetch(`https://viacep.com.br/ws/${props.cep}/json/`)
      .then(response => response.json())
      .then(data => {
        setData(data);
        setError(false);
        props.onSuccess(data);
      })
      .catch(err => {
        setError(true);
      })
      .finally(() => setLoading(false));
  }

  return (
    props.children({
      loading,
      data,
      error,
      fetch: getCep
    }) || null
  );
}

ViaCepHook.propTypes = {
  cep: PropTypes.string.isRequired,
  lazy: PropTypes.bool,
  onSuccess: PropTypes.func
};

export default ViaCepHook;
