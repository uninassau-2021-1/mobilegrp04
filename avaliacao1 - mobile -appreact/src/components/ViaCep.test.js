import React from "react";
import renderer from "react-test-renderer";
import ViaCep from "./ViaCep";

describe("ViaCep", () => {
  it("snapshot renders", () => {
    const component = renderer.create(<ViaCep cep="">{() => {}}</ViaCep>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});
