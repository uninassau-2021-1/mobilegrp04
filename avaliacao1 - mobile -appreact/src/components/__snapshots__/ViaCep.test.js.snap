// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViaCep snapshot renders 1`] = `null`;
